src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.spec.ts

```diff
@@ -518,5 +519,24 @@ describe('Component: GeneComparisonToolComponent', () => {
       const expected2 = 'ENSG00000147065';
       expect(label2).toBe(expected2);
     });
+
+    it('should set circle size to zero for null/undefined pValues', () => {
+      let tissue: GCTGeneTissue | undefined;
+      // null/undefined values should be zero
+      const result = component.getCircleSize(tissue?.adj_p_val);
+      expect(result).toBe(0);
+    });
```

Comment:
Since `null` caused this bug, would it make sense to explicitly test the `null` case as well?

```suggestion
    it('should set circle size to zero for undefined pValues', () => {
      let tissue: GCTGeneTissue | undefined;
      // undefined values should be zero
      const result = component.getCircleSize(tissue?.adj_p_val);
      expect(result).toBe(0);
    });
    
    it('should set circle size to zero for null pValues', () => {
      let tissue: GCTGeneTissue = {
        name: "TCX",
        logfc: null,
        adj_p_val: null,
        ci_l: null,
        ci_r: null
      };
      // null values should be zero
      const result = component.getCircleSize(tissue.adj_p_val);
      expect(result).toBe(0);
    });
```

Reply:
You're right, I will break this out into two tests